### PR TITLE
Add close command to MSFTbot config

### LINKED ISSUE #55005: Moderators should have a way to close PRs.
### Description of the new feature/enhancement

Continuing the theme of https://github.com/microsoft/winget-pkgs/issues/35372 (which I hear is very close!), Moderators should have a way to request wingetbot/msftbot close a PR given a reason. Many times we close PRs by marking them as being "resolved" by another PR, which just [causes confusion](https://github.com/microsoft/winget-pkgs/pull/54700#issuecomment-1072990783) unless everyone is aware of the workaround.

There may be a way to do this via a GitHub Team (delegating the power to close PRs), but I don't know if you can without consuming licenses (which is obviously a no-go for outside people). I know that was talked about at one stage, but I guess it was dropped. As such, I think the bot should just take care of it on our behalf.

### Proposed technical implementation details (optional)

"@wingetbot close" should close a PR. 

Preferably, it would refuse to do so if we hadn't made any other comments on the PR (or if there was no other text in the comment with the command), forcing us to explain why it was being closed. Although we're usually good about explaining, having a little enforcement wouldn't be a bad thing.

### PR DESCRIPTION
**Pull request by Trenly**
Allows moderators to close pull requests using `Close with reason <reason>;`

```
PS C:\> $jsonString = '"Close\\s+with\\s+reason\\s*:[\\w\\s\\-\\(\\)\\[\\]\\{\\}\\\\\\/.+=@#$%&^*`~|\",<>?]*(?=;)"'
PS C:\> $regex = $jsonString|ConvertFrom-Json
PS C:\> $regex
Close\s+with\s+reason\s*:[\w\s\-\(\)\[\]\{\}\\\/.+=@#$%&^*`~|",<>?]*(?=;)
PS C:\> 'Close With Reason:  Test' -match $regex
False
PS C:\> 'Close With Reason:  Test;' -match $regex
True
PS C:\> 'Close With Reason:  Test
>>
>> with returns;' -match $regex
True
PS C:\> 'Close With Reason:  Test
>>
>> with returns
>> and special chars
>> %^&*(){}[];' -match $regex
True
PS C:\WINDOWS\>
```

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/66249)

### CODE REVIEW

**Comment by denelon** [comment]
I'm assuming it's just the "enforcement mentioned by @jedieaston in the related issue:
* #55005

File: .github/fabricbot.json [hunk 1]
```diff
@@ -1628,6 +1628,122 @@
       },
       "disabled": false
     },
+    {
+      "taskType": "trigger",
+      "capabilityId": "IssueResponder",
+      "subCapability": "IssueCommentResponder",
```
Comment (denelon) [inline .github/fabricbot.json:1634]:
```suggestion
      "subCapability": "PullRequestCommentResponder",
```

**Comment by Trenly** [comment]
> What reason(s) are you considering?
> 
> regex always makes me nervous :)

It looks for “any character” except ; which is the terminating character

Breakdown of the regex is:
* Begins with “Close”
* Has some number of whitespace characters greater than 0
* Second word is “with”
* Has some number of whitespace characters greater than 0
* Third word is “reason”
* Next chracter is “:”
* Following characters are any word, any whitespace character, or any character in the set -(){}[]\/.+=@#$%&^*`~|",<>? repeated from 0 to infinitely many times
* Terminated by a semicolon

**Comment by denelon** [comment]
I merged:
* #66250 

so this might have a merge conflict.

**Comment by Trenly** [comment]
> I merged:
> 
> * [Automatically Assign PR's with Author-Not-Authorized label #66250](https://github.com/microsoft/winget-pkgs/pull/66250)
> 
> so this might have a merge conflict.

If it does I’ll rebase

**Review by denelon — APPROVED** [review]

**Pull request merged**